### PR TITLE
gh-365: upload HTML rendering of notebooks as artifact

### PR DESCRIPTION
**Pull request by Saransh-cpp**
The workflows now produce an HTML rendering of the notebook and upload it as artifact. I have tweaked the nox session to help with this. All the notebooks have been rerun to update the outputs with the recent changes.

Closes: #365

### CODE REVIEW

**Comment by ntessore** [comment]
Great work @Saransh-cpp! Can you revert the changes to the .ipynb files, which will be nothing but updated timestamps and redrawn plots?

File: .github/workflows/examples.yml [hunk 1]
```diff
@@ -45,3 +45,9 @@ jobs:
         run: pip install nox
       - name: Run examples
         run: nox -s examples
+      - name: Generate HTML
+        run: nox -s examples -- html
+      - uses: actions/upload-artifact@v4
+        with:
+          name: Examples
+          path: examples/**/*.html
```
Comment (ntessore) [inline .github/workflows/examples.yml:53]: I think this now executes all of the examples twice: once with `jupyter execute examples/**/*.ipynb` and once with `jupyter nbconvert --execute`. You could either remove the first step, or I believe you can remove `--execute` from the second step.

**Comment by ntessore** [comment]
Could you add a dummy file somewhere that triggers the "Test examples" step, to be removed before merging?

**Comment by paddyroddy** [comment]
> Could we add this to `pre-commit`? This keeps the plots, and removes all metadata which doesn't actually get displayed on [glass.readthedocs.io](https://glass.readthedocs.io) anyway
> 
> ```yaml
> repos:
>   - repo: https://github.com/kynan/nbstripout
>     rev: 0.7.1
>     hooks:
>       - id: nbstripout
>         args:
>           - --keep-output
> ```

Have added this and raised https://github.com/glass-dev/glass/pull/380

**Comment by Saransh-cpp** [comment]
The artifact upload worked - https://github.com/glass-dev/glass/actions/runs/11701933760

File: glass/remove_me.py [hunk 2]
```diff
@@ -0,0 +1 @@
+"""to test example artifacts."""
```
Comment (paddyroddy) [inline glass/remove_me.py:1]: Should this be removed?

File: .github/workflows/examples.yml [hunk 3]
```diff
@@ -52,3 +52,11 @@ jobs:
...
         env:
           FORCE_COLOR: 1
+      - name: Generate HTML
+        run: nox -s examples -- html
+        env:
+          FORCE_COLOR: 1
+      - uses: actions/upload-artifact@v4
+        with:
+          name: Examples
+          path: examples/**/*.html
```
Comment (paddyroddy) [inline .github/workflows/examples.yml:62]: Just to keep in line with #388
```suggestion

      - name: Generate HTML
        run: nox -s examples -- html
        env:
          FORCE_COLOR: 1

      - name: Upload artefact
        uses: actions/upload-artifact@v4
        with:
          name: Examples
          path: examples/**/*.html
```

**Review by paddyroddy — CHANGES_REQUESTED** [review]

File: glass/remove_me.py (hunk 2 above)
Comment (Saransh-cpp) [reply glass/remove_me.py]: Yes, was added to trigger the tests.

**Review by paddyroddy — APPROVED** [review]
Looks good!

**Pull request merged**